docs/user-guide/java-lib.rst

```diff
@@ -0,0 +1,220 @@
+GraphAr Java Library
+====================
+
+Overview
+--------
+
+Based on an efficient FFI for Java and C++ called
+`fastFFI <https://github.com/alibaba/fastFFI>`__, the GraphAr Java
+library allows users to write Java for generating, loading and
+transforming GAR files. It consists of several components:
+
+-  **Information Classes**: As same with in the C++ library, the
+   information classes are implemented to construct and access the meta
+   information about the **graphs**, **vertices** and **edges** in
+   GraphAr.
+
+-  **Writers**: The GraphAr Java writer provides a set of interfaces
+   that can be used to write Apache Arrow VectorSchemaRoot into GAR
+   files. Every time it takes a VectorSchemaRoot as the logical table
+   for a type of vertices or edges, then convert it to ArrowTable, and
+   then dumps it to standard GAR files (CSV, ORC or Parquet files) under
+   the specific directory path.
+
+-  **Readers**: The GraphAr Java reader provides a set of interfaces
+   that can be used to read GAR files. It reads a collection of vertices
+   or edges at a time and assembles the result into the ArrowTable.
+   Similar with the reader in the C++ library, it supports the users to
+   specify the data they need, e.g., reading a single property group
+   instead of all properties.
+
+Get GraphAr Java Library
+------------------------
+
+Building from source
+~~~~~~~~~~~~~~~~~~~~
+
+Only support installing from source currently, but we will support
+installing from Maven in the future.
+
+Firstly, install llvm-11. ``LLVM11_HOME`` should point to the home of
+LLVM 11. In Ubuntu, it is at ``/usr/lib/llvm-11``. Basically, the build
+procedure the following binary:
+
+-  ``$LLVM11_HOME/bin/clang++``
+
+-  ``$LLVM11_HOME/bin/ld.lld``
+
+-  ``$LLVM11_HOME/lib/cmake/llvm``
+
+Tips:
+
+-  From apt-get:
+
+   .. code-block:: bash
+
+      $ sudo apt-get install llvm-11 clang-11 lld-11 libclang-11-dev libz-dev -y
+      $ export LLVM11_HOME=/usr/lib/llvm-11
+
+-  Compile from source with this
+   `script <https://github.com/alibaba/fastFFI/blob/main/docker/install-llvm11.sh>`__:
+
+   .. code-block:: bash
+
+      $ export LLVM11_HOME=/usr/lib/llvm-11
+      $ export LLVM_VAR=11.0.0
+      $ sudo ./install-llvm11.sh
+
+Make the graphar-java-library directory as the current working
+directory:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/alibaba/GraphAr.git
+    $ cd GraphAr
+    $ git submodule update --init
+    $ cd java
+
+Compile package:
+
+.. code-block:: bash
+
+    $ mvn clean install -DskipTests
+
+Then set GraphAr as a dependency in maven project:
```

Comment:
the maven project has not set, so maybe we should not include this part. We can add this part back if maven is ready

Reply:
User needs to add this part in pom.xml so the project can import the gar-java library

Reply:
You mean after building the project, the user can use the project directly?